hr_contract_cae/models/hr_contract.py

```diff
@@ -18,21 +89,108 @@ class Contract(models.Model):
+                            if contract.initial_contract_id.type_id.name
+                            and contract.index_amendment > 0
+                            else ""
+                        ),
+                        (contract.type_id.name or ""),
+                    ],
+                )
+            )
+
+    @api.multi
+    @api.depends("index_amendment")
+    def _compute_type_echelon(self):
+        for contract in self:
+            contract.type_echelon = (
+                "main" if contract.index_amendment == 0 else "amendment"
+            )
 
     @api.multi
     @api.depends("hours", "hourly_wage")
     def _compute_wage(self):
         for contract in self:
-            contract.wage = self.hours * self.hourly_wage
+            contract.wage = contract.hours * contract.hourly_wage
 
     @api.onchange("date_start", "duration")
     def onchange_date_start_duration(self):
         if self.date_start and self.duration:
-            self.date_end = self.date_start + relativedelta(months=self.duration)
+            self.date_end = self.date_start + relativedelta(
+                months=self.duration
+            )
 
     @api.onchange("date_end")
     def onchange_date_end(self):
         if self.date_start and self.date_end:
             rd = relativedelta(self.date_end, self.date_start)
             self.duration = rd.months + rd.years * 12
+
+    @api.multi
+    def create_amendment(self):
+        self.ensure_one()
+        latest_contract_id = self.latest_contract_id
+        contract = latest_contract_id.copy(
+            {
+                "state": "draft",
+                "type_id": self.env["hr.contract.type"]
+                .search([("echelon", "=", "amendment")], limit=1)
+                .id,
+                "initial_contract_id": latest_contract_id.initial_contract_id.id,
+                "parent_contract_id": latest_contract_id.id,
+                "index_amendment": latest_contract_id.index_amendment + 1,
+            }
+        )
+        latest_contract_id.child_contract_id = contract
+
+        return {
+            "type": "ir.actions.act_window",
+            "res_model": "hr.contract",
+            "view_mode": "form",
+            "res_id": contract.id,
+            "target": "current",
+            "context": {"form_view_initial_mode": "edit"},
+        }
+
+    @api.multi
+    @api.onchange("initial_contract_id", "type_id")
+    def _inital_to_latest(self):
+        self.ensure_one()
+        current_contract_id = self.initial_contract_id
+        count_type = current_contract_id.type_id == self.type_id
+        while current_contract_id.child_contract_id:
+            count_type += (
+                current_contract_id.child_contract_id.type_id == self.type_id
+            )
+            current_contract_id = current_contract_id.child_contract_id
+        self.latest_contract_id = current_contract_id
+        self.count_type = count_type
```

Comment:
`count_type`: you're assigning a boolean to an integer and adding boolean.

Reply:
Would this give errors or is it just non-zen-of-Python?
```python
a = 2
a += True
# a = 3
# seems to work...
```

Reply:
it's not pythonesque according to me. When thinking about code, operations on integer should give integer and operation on boolean should give booleans.

Reply:
so you'd prefer to cast to boolean first?

Reply:
if you're doing addition i'd rather cast to integers. Or more simply:

```
while current_contract_id.child_contract_id:
    if current_contract_id.child_contract_id.type_id == self.type_id:
            count_type += 1
```